### ISSUE TITLE
Wrong tax location when adding products to orders via admin (AJAX)

### ISSUE THREAD

**Issue by vendidero** [woocommerce/woocommerce#30231]
Hi there,

when adding products to an admin order an AJAX request `WC_AJAX::add_order_item` is fired. This calls `WC_AJAX::maybe_add_order_item` which eventually calls `WC_Abstract_Order::add_product`. By default the total is calculated by `wc_get_price_excluding_tax`. In case products include tax, taxes are being calculated to retrieve the net price. By default the base tax rates are used. But if you've added the filter:

`add_filter( 'woocommerce_adjust_non_base_location_prices', '__return_false' );`

to keep gross prices fixed no matter the country chosen, tax rates returned by `WC_Tax::get_rates` are used. As no customer object has been passed to the method explicitly, the main `WC_Customer` object (loaded from the session while performing the AJAX request) will be used - that's the current admin user. That's why the tax rates are chosen based on the current logged in users' address (`WC_Tax::get_tax_location`) instead of based on the current order data.

Expected: In case the `woocommerce_adjust_non_base_location_prices` filter is used to keep gross prices static when adding a new product via the admin order screen the net price should be calculated based on current order data (passed to the AJAX request).

Actual: The tax rates are chosen based on the current logged in admin user.

Solution: We will probably need to pass additional arguments (e.g. total, subtotal) calculated based on the order data explicitly to `WC_Abstract_Order::add_product` - https://github.com/woocommerce/woocommerce/blob/trunk/includes/class-wc-ajax.php#L958 

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

Tested with the latest RC2 of Woo 5.5 and latest WP.

Cheers

**Comment by juliaamosova** [comment]
Hi @vendidero,

Thank you for opening the issue! It requires further feedback from the WooCommerce Core team. I am adding the `needs developer feedback` label to this issue so that the Core team could take a look. 

Please note it may take a few days for them to get to this issue. Thank you for your patience.

**Comment by roykho** [comment]
Hi @vendidero - Thanks for reporting this. To further investigate, could you please give us the full detail step by step setup to reproduce this issue? Because when it comes to taxes, there are many different settings in WooCommerce which could possibly affect how the calculations are done.

Thanks again!

**Comment by vendidero** [comment]
Hi @roykho,

actually the tax setup doesn't matter that much (as the issue is related to taxes being calculated based on the logged in customer/user location and not the order location) but here you go:

1. Enable prices including taxes
2. Calculate taxes based on customers shipping address
3. Insert a tax rate of 19 % for Germany and a tax rate of 21 % for Portugal
4. Make sure that your admin user has Germany selected as shipping/billing country
5. Create an order (manual or via checkout) with shipping country Portugal (and 21 % taxes)
6. Add a product to the order - see the net price being calculated based on 19 % instead of 21 %.

Cheers

**Comment by roykho** [comment]
Hi @vendidero - I am assuming I have to choose the `admin` as the customer such as here ![](https://d.pr/i/3baVsQ+) and also use the filter `add_filter( 'woocommerce_adjust_non_base_location_prices', '__return_false' );` to be able to replicate this?

**Comment by vendidero** [comment]
Hi,

no, just leave it blank. The admin doesn't need to be the customer - please check my explanation of the issue in the first place and check the code - the global WC_Customer object is being used (which is the current logged in user) to determine tax location. Yes, `add_filter( 'woocommerce_adjust_non_base_location_prices', '__return_false' );` is necessary.

Cheers

**Comment by roykho** [comment]
Hi @vendidero - so I've tried to reproduce this based on your steps but everything seems to be working on my side. Please have a look at the screenshots.

Profile:
![](https://d.pr/i/Sj3y1Y+)

Tax rates:
![](https://d.pr/i/LrlU3x+)

Order with Portugal:
![](https://d.pr/i/JKCjXj+)

Order with Germany:
![](https://d.pr/i/DuqqPi+)

In my testing it does use what is provided by the order's address and not from admin's address. My colleague also tested this based on your description and concluded the same as I did. Are we missing any steps?

**Comment by vendidero** [comment]
Hi,

seems like you did not enable prices including tax within the tax settings? Did you enable the filter as well?

Cheers

**Comment by roykho** [comment]
Hi @vendidero - yes we've tested both ways and added the filter per your instructions.

**Comment by vendidero** [comment]
Hi there,

the issue does not occur when recalculating taxes and/or adding taxes but when adding a new line item to a newly created order (make sure to save the order after updating the order location data, e.g. shipping/billing address). I've just tested again and the net price calculated after adding a line item is calculated based on the tax rates linked to the logged in user and not based on order data. The issue is not tax calculation but tax calculation while determining the net price for a product which is taxable. This is the logical consequence of calling `wc_get_price_excluding_tax` during the AJAX request which gets the rates via `WC_Tax::get_rates()` when the `woocommerce_adjust_non_base_location_prices` filter is set to `false`. 

See: https://github.com/woocommerce/woocommerce/blob/trunk/includes/wc-product-functions.php#L1061 and https://github.com/woocommerce/woocommerce/blob/50c8b7d1ebdf21c7484a1e98a7f31e2e995c700d/includes/class-wc-tax.php#L483

Cheers

**Comment by roykho** [comment]
@vendidero - Thanks for the additional information. I can now reproduce the issue. I have flagged this as a bug. We might not be able to get to this right away but will be worked on soonest.
